src/cmd/ksh93/fun/dirs

```diff
@@ -55,6 +69,13 @@ function _cd
     case $dir in
     \~*)   dir=$HOME${dir#\~}
     esac
+    # mcd passes escaped ~, which must be dealt with
+    if     ((_mcd_called>0))
+    then   case "$@" in
```

Comment:
Using `"$@"` in this context makes no sense as that quoted expansion generates one word per positional parameter, but only exactly one word is possible between `case` and `in`, so the behaviour of `"$@"` is unspecified in this context. On ksh, in practice it's the same as `"$*"` here, i.e., the positional parameters get joined into one string separated by the first character of `$IFS`. But in this case you're only checking the first character of the first parameter, so you might as well just use `$1` (no quotes needed).

Reply:
Yes, only the first parameter would be wanted here.